### PR TITLE
Set cumulative time interval different from update period [12738]

### PR DESCRIPTION
**Pull request by rsanchez15**
Calculates the cumulative statistic with respect to a time interval other than the update period.

### CODE REVIEW

**Comment by juanlofer-eprosima** [comment]
Perhaps a prefix could be added to the labels of cumulative series.

File: qml/DynamicDisplayStatisticsDialog.qml [hunk 1]
```diff
@@ -306,6 +487,25 @@ Dialog {
...
+            } else {
+                var cumulative_minutes = cumulativeIntervalToMinutes()
+                var cumulative_only_seconds = (cumulativeIntervalSeconds.text === "") ? 0 : parseInt(cumulativeIntervalSeconds.text)
+
+                if (Number.isInteger(cumulative_minutes)) {
+                    text += "_cumulative-" + cumulative_minutes + "min"
+                } else if (Math.floor(cumulative_minutes) == 0) {
+                    text += "_cumulative-" + cumulativeIntervalToSeconds() + "s"
+                } else {
+                    text += "_cumulative-" + Math.floor(cumulative_minutes) + "min," + cumulative_only_seconds + "s"
```
Comment (juanlofer-eprosima) [inline qml/DynamicDisplayStatisticsDialog.qml:504]:
```suggestion
                    text += "_cumulative-" + Math.floor(cumulative_minutes) + "min," + cumulative_only_seconds % 60 + "s"
```

File: include/fastdds_monitor/backend/SyncBackendConnection.h [hunk 2]
```diff
@@ -222,6 +230,18 @@ class SyncBackendConnection
     bool data_kind_has_target(
             const DataKind& data_kind);
 
+    /*
```
Comment (juanlofer-eprosima) [inline include/fastdds_monitor/backend/SyncBackendConnection.h:233]:
```suggestion
    /**
```

File: qml/DynamicDisplayStatisticsDialog.qml [hunk 3]
```diff
@@ -24,6 +24,7 @@ import QtQuick.Controls 2.15
 import QtQuick.Controls 1.4 as QCC1
 import QtQuick.Controls.Styles 1.4
 import QtQml.Models 2.12
+import Theme 1.0
```
Comment (juanlofer-eprosima) [inline qml/DynamicDisplayStatisticsDialog.qml:27]: Sort alphabetically?

File: src/backend/SyncBackendConnection.cpp [hunk 4]
```diff
@@ -624,37 +625,22 @@ std::vector<StatisticsData> SyncBackendConnection::get_data(
...
-                two_entities_data = true;
-                if (!target_entity_id.is_valid())
-                {
-                    target_entity_id = source_entity_id;
-                }
-                auto target_ids_tmp = get_entities(kinds_supported.second, target_entity_id);
-                target_ids.insert(target_ids.end(), target_ids_tmp.begin(), target_ids_tmp.end());
-            }
-        }
+    if (source_ids.empty())
```
Comment (juanlofer-eprosima) [inline src/backend/SyncBackendConnection.cpp:637]: You could maybe print a warning.

File: src/backend/SyncBackendConnection.cpp [hunk 5]
```diff
@@ -695,6 +681,65 @@ std::vector<StatisticsData> SyncBackendConnection::get_data(
...
+            }
+        }
+    }
+    catch (const Exception& e)
+    {
+        qWarning() << "Fail getting data: " << e.what();
+        static_cast<void>(e); // In release qWarning does not compile and so e is not used
+
+        return false;
+    }
```
Comment (juanlofer-eprosima) [inline src/backend/SyncBackendConnection.cpp:738]: This exception does not need to be handled here, it is done in get_data.

**Review by juanlofer-eprosima — CHANGES_REQUESTED** [review]

File: src/backend/SyncBackendConnection.cpp (hunk 5 above)
Comment (rsanchez15) [reply src/backend/SyncBackendConnection.cpp:738]: Yes, this is needed as there is a call to a backend function (`StatisticsBackend::get_data_supported_entity_kinds()`). 
However, I've modified the logging message.

File: qml/DynamicDisplayStatisticsDialog.qml (hunk 3 above)
Comment (rsanchez15) [reply qml/DynamicDisplayStatisticsDialog.qml:27]: We don't sort the imports in qml files

**Review by juanlofer-eprosima — APPROVED** [review]
LGTM

**Pull request merged**